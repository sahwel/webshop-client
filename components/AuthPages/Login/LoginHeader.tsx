import React from 'react';

const LoginHeader = () => {
  return (
    <div className="select-none">
      <h1 className="text-[300%]">Let’s sign you in!</h1>
      <h3>Welcome back,</h3>
      <h3>you’ve been missed!</h3>
    </div>
  );
};

export default LoginHeader;
