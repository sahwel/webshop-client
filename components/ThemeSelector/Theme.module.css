
@tailwind components;

@layer components {
    .theme {
       @apply rounded-full p-2  flex justify-center items-center
    }
}