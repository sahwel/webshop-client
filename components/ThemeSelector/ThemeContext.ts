/* eslint-disable no-undef */
import { createContext } from 'react';
import { Themes } from './definitions';

export interface ThemeContextModel {
  setTheme(): void;
  getTheme(): Themes;
}

export const ThemeContext = createContext<ThemeContextModel>(null as any);
