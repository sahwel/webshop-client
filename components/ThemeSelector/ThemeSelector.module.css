
@tailwind components;

@layer components {
    .theme_sececltor {
        @apply grid grid-cols-2 z-50 cursor-pointer bg-white absolute right-5  rounded-full border-2 border-theme-border-light
    }  
    
    .theme_sun{
        @apply text-orange-600 
    }

    .theme_moon{
        @apply text-gray-400 rotate-[255deg]
    }

    .theme_moon_highlited{
        @apply bg-[#303841]
    }

    .theme_sun_highlited{
        @apply bg-orange-200
    }
}