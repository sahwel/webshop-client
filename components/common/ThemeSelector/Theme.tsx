import { FunctionComponent, ReactNode } from 'react';
import cl from 'classnames';
import styles from './theme.module.css';

interface ThemeProps {
  children?: ReactNode;
  isHighlighted: boolean;
  highlightedClass: string;
}

const Theme: FunctionComponent<ThemeProps> = ({
  children,
  isHighlighted,
  highlightedClass,
}) => {
  return (
    <div className={cl({ [highlightedClass]: isHighlighted }, styles.theme)}>
      {children}
    </div>
  );
};

export default Theme;
