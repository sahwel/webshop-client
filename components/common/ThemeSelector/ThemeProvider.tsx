import {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useLayoutEffect,
  useMemo,
  useState,
} from 'react';
import { ThemeContext, ThemeContextModel } from './ThemeContext';
import { Themes } from './definitions';

interface ThemeProviderProps {
  children?: ReactNode;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({ children }) => {
  const [currentTheme, setCurrentTheme] = useState<Themes>('light');

  useLayoutEffect(() => {
    // todo: get alternative solution
    const theme = localStorage.theme ? localStorage.theme : 'light';

    if (theme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    setCurrentTheme(theme);

    return () => {};
  }, []);
  const contextValue: ThemeContextModel = useMemo(
    () => ({
      setTheme() {
        setCurrentTheme((oldTheme) => {
          const newTheme = oldTheme === 'light' ? 'dark' : 'light';

          if (newTheme === 'dark') {
            document.documentElement.classList.add('dark');
          } else {
            document.documentElement.classList.remove('dark');
          }
          localStorage.theme = newTheme;
          return newTheme;
        });
      },
      getTheme() {
        return currentTheme;
      },
    }),
    [currentTheme]
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;

export function useTheme() {
  const value = useContext(ThemeContext);
  return value;
}
