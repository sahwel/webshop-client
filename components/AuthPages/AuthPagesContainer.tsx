import Image from 'next/image';
import { FunctionComponent, ReactNode } from 'react';
import { useTheme } from '../common/ThemeSelector/ThemeProvider';
import styles from './AuthPagesContainer.module.css';
import darkBlob from '../../public/blobs/dark/login-register/blob.svg';
import lightBlob from '../../public/blobs/light/login-register/blob.svg';
import ThemeSelector from '../common/ThemeSelector/ThemeSelector';
interface AuthPagesContainerProps {
  children: ReactNode;
}

const AuthPagesContainer: FunctionComponent<AuthPagesContainerProps> = ({
  children,
}) => {
  const theme = useTheme();

  return (
    <>
      <ThemeSelector />
      <div className={styles.auth_pages_container}>
        <div className={styles.blob_container}>
          <Image
            className={styles.blob}
            src={theme.getTheme() === 'dark' ? darkBlob : lightBlob}
            alt="blob"
            priority={true}
            layout="fill"
          />
        </div>
        {children}
      </div>
    </>
  );
};

export default AuthPagesContainer;
