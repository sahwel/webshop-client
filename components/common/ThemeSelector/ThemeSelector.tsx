import { SunIcon } from '@heroicons/react/outline';
import { MoonIcon } from '@heroicons/react/solid';
import { useCallback, useLayoutEffect, useState } from 'react';
import cl from 'classnames';
import Theme from './Theme';
import styles from './ThemeSelector.module.css';
import { useTheme } from './ThemeProvider';

const ThemeSelector = () => {
  const siteTheme = useTheme();
  const handleChange = useCallback(() => {
    siteTheme.setTheme();
  }, []);

  return (
    <div className={styles.theme_sececltor} onClick={handleChange}>
      <Theme
        highlightedClass={styles.theme_sun_highlited}
        isHighlighted={siteTheme.getTheme() === 'light'}
      >
        <SunIcon className={cl('icon-6', styles.theme_sun)} />
      </Theme>

      <Theme
        highlightedClass={styles.theme_moon_highlited}
        isHighlighted={siteTheme.getTheme() === 'dark'}
      >
        <MoonIcon
          className={cl('icon-6', styles.theme_moon, {
            ['!text-white ']: siteTheme.getTheme() === 'dark',
          })}
        />
      </Theme>
    </div>
  );
};

export default ThemeSelector;
