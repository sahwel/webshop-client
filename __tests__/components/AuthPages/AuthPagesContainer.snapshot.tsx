import { render, renderHook, RenderHookResult } from '@testing-library/react';
import LoginHeader from '@/components/AuthPages/Login/LoginHeader';
import AuthPagesContainer from '@/components/AuthPages/AuthPagesContainer';
import { useTheme } from '@/components/common/ThemeSelector/ThemeProvider';
import {
  mockUseThemeSelector,
  themeSelectorWrapper,
} from '../../../testUtils/MockThemeSelector';
import React from 'react';
React.useContext = mockUseThemeSelector;

describe('AuhtPahtesContainer Snapshot', () => {
  let result: RenderHookResult<any, any>;
  beforeEach(() => {
    result = renderHook(() => useTheme(), {
      wrapper: themeSelectorWrapper,
    });
  });

  it('renders LoginHeader unchanged', () => {
    const { container } = render(<AuthPagesContainer children={'test'} />);
    expect(container).toMatchSnapshot();
  });
});
