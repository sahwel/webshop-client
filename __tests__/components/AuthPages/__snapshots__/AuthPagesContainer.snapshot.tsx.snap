// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuhtPahtesContainer Snapshot renders LoginHeader unchanged 1`] = `
<div>
  <div
    class="auth_pages_container"
  >
    <div
      class="blob_container"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="blob"
          class="blob"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/blobs/light/login-register/blob.svg"
          srcset="/blobs/light/login-register/blob.svg 640w, /blobs/light/login-register/blob.svg 750w, /blobs/light/login-register/blob.svg 828w, /blobs/light/login-register/blob.svg 1080w, /blobs/light/login-register/blob.svg 1200w, /blobs/light/login-register/blob.svg 1920w, /blobs/light/login-register/blob.svg 2048w, /blobs/light/login-register/blob.svg 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
    </div>
    test
  </div>
</div>
`;
